### PR TITLE
TRITON-2517 create generic diskio plugin for global zones

### PR DESCRIPTION
**Pull request by jsadowyj**
## Overview
Removes the `nvme_diskio` and replaces it with the generic `diskio` plugin, which exposes metrics for both NVME and SATA/SAS devices.

## AI Disclosure
Claude generated most of this script, and I made modifications.

### cmon logs
This was tested on one of our CNs in DA11 with only SAS drives. 

`cmon-agent` logs do not return any errors:
```
[root@s01 (us-central-1a) /opt/custom/cmon/gz-plugins]# bunyan -l warn $(svcs -L cmon-agent)
[ Oct 28 17:32:54 Stopping because service restarting. ]
[ Oct 28 17:32:54 Executing stop method (:kill). ]
[ Oct 28 17:32:54 Executing start method ("/usr/bin/ctrun -l child -o noorphan /opt/smartdc/agents/lib/node_modules/cmon-agent/node/bin/node --abort_on_uncaught_exception /opt/smartdc/agents/lib/node_modules/cmon-agent/bin/cmon-agent &"). ]
[ Oct 28 17:32:54 Method "start" exited with status 0. ]
```

### grafana
The metrics are being visualized correctly in Grafana:

image

### edge case

This was also tested on @danmcd CoAL deployment, which handles the edge cast of having BOTH NVME and SAS/SATA drives. This case is what made it into the mock test in this repo.

### CODE REVIEW

File: gz-plugins/diskio.prom [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                 "sd::/^sd[0-9]+$/:wtime" 2>/dev/null)
+
+# shellcheck disable=SC2181
+if [ -z "$stats" ]; then
+    # If kstat fails or returns no devices, exit silently
+    exit 0
+fi
+
+
+# Process all stats with AWK
```
Comment (danmcd) [inline gz-plugins/diskio.prom:39]: Probably at this point in time a NOP, but the PI needs to be release-20200423 because that's when awk(1) became the same as nawk(1) instead of oawk(1).  See illumos#12482 .

**Review by danmcd — COMMENTED** [review]
Checkpointing.  Also, does this PR need RFD 187 checking/credit?

File: gz-plugins/diskio.prom (hunk 1 above)
Comment (danmcd) [reply gz-plugins/diskio.prom:39]: Going to resolve this, but if someone uses these on a PI that's EARLIER than 20200423, it's their own fault.

Comment (jsadowyj) [reply gz-plugins/diskio.prom:39]: We specifically mention in this repo that these are unofficial.

Per README.md
> These are unofficial. They were previously used in Joyent's production infrastructure, and given new life in Edgecast's production infrastructure. These plugins may or may not work with your systems. A valid workaround for you might be to install to an alternate location and symlink the plug-ins you wish to use.

I would like to see this functionality make it into the actual CMON agent at some point, because I'd argue that these are metrics most users would want. If we go that route, it would be written in JS, so we wouldn't have to worry about AWK/Bash compatibility.

**Review by danmcd — COMMENTED** [review]
This needs release-20200423 (as do other better-use-of-awk(1) ones).

Also as I said on another PR: Before I approve this, a couple of quick questions:

1.) The single-awk-run was Claude-assisted? If so, the assisted-by line needs to go into the commit.

2.) But it was suggested by @jperkin ? If so, a Portions-by-jperkin line needs to go into the comment as well.

**Review by danmcd — APPROVED** [review]
Questions answered offline, and will be reflected in the squash-and-merge commit message.
